rotkehlchen/chain/manager.py

```diff
@@ -621,10 +621,18 @@ def query_balances(
             self.query_polkadot_balances(ignore_cache=ignore_cache)
         if should_query_avax:
             self.query_avax_balances(ignore_cache=ignore_cache)
+        if ignore_cache is True:
+            self.derive_new_addresses_from_xpubs()
 
         self.totals = self.balances.recalculate_totals()
         return self.get_balances_update()
 
+    def derive_new_addresses_from_xpubs(self) -> None:
+        """Derive new addresses from xpubs."""
+        xpub_manager = XpubManager(chain_manager=self)
+        xpub_manager.check_for_new_xpub_addresses(blockchain=SupportedBlockchain.BITCOIN)
+        xpub_manager.check_for_new_xpub_addresses(blockchain=SupportedBlockchain.BITCOIN_CASH)
```

Comment:
I believe it would be better to not combine this two. Correct me if I'm wrong but the derivation process makes some external calls and those can be slow. If you are querying only bch balances you would also be deriving BTC addresses and that might not be needed. Also think in accounts with a lot of addresses

Reply:
Yes, you're right about that. Fixing atm